### PR TITLE
feat: add concurrency

### LINKED ISSUE #2: [suggestion] Enable concurrency
Perhaps something like this:
```ts
await P(items, { concurrency: 20 })
```
To establish hard limits on concurrency!

### PR DESCRIPTION
**Pull request by henrycunh**
Added one dependency (`p-limit`), is that a problem? 

Would rather have a pure implementation of concurrency control?

Closes #2

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
+      const limit = pLimit(this.options.concurrency)
+      solved = Promise.all(items.map(p => limit(() => p)))
+    }
+    else { solved = Promise.all(items) }
+
+    return solved.then(l => l.filter((i: any) => i !== NULL))
   }
 
-  constructor(public items: Iterable<T> = []) {
+  constructor(public items: Iterable<T> = [], options?: POptions) {
```
Comment (antfu) [inline src/index.ts]:
```suggestion
  constructor(public items: Iterable<T> = [], public options?: POptions) {
```

Also need to pass the options to `map` and `filter`

Comment (henrycunh) [reply src/index.ts]: done :]

**Comment by antfu** [comment]
Eventually, I would like to inline p-limit, but for now I think it's fine to have it as a dep

**Comment by henrycunh** [comment]
just saw tests failed because the runner is 1ms faster than the lower bound i've set for the test to run, so fast!

will relax it even further

**Comment by antfu** [comment]
Tests failing, I guess we will need a more stable time-irrelevant test method

**Comment by henrycunh** [comment]
> Tests failing, I guess we will need a more stable time-irrelevant test method

agreed!

**Pull request merged**